benchmark/pkg/scraper/scraper.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scraper
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"strings"
+
+	"k8s.io/contrib/test-utils/utils"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	"github.com/golang/glog"
+)
+
+const (
+	readingJunk     			= iota
+	readingAPICallLatencies		= iota
+	readingPodStartupLatency 	= iota
+	buildLogPath 				= "build-log.txt"
+	testNameIdentifierTag		= "[BeforeEach] [k8s.io]"
+	apiLatencyResultTag			= "API calls latencies: "
+	podLatencyResultTag			= "Pod startup latency: "
+)
+
+// This function takes a job name and a run number, gets that run's build log,
+// reads it and parses JSON summaries of API call and Pod startup latencies.
+func GetMetricsForRun(job string, run int) (map[string]*e2e.APIResponsiveness, map[string]*e2e.PodStartupLatency, error) {
```

Comment:
Yes, you've got the right feeling. There was one such function in your compare tool (Ref: https://github.com/kubernetes/perf-tests/blob/master/compare/src/scraper.go#L39), however we aren't using it for 3 reasons:
- It doesn't work for kubemark logs where we don't have the metrics printed b/w those delimiters (used in the function)
- We want api and pod latencies, while that function gives logs size summary, resource usage summary and metrics summary (which includes api latencies but not pod startup latencies)
- This has more cleaner state machine for parsing which uniformly works for all the metrics we want